azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/participantlist/ParticipantListViewModel.kt

```diff
@@ -27,26 +27,41 @@ internal class ParticipantListViewModel {
         return displayParticipantListStateFlow
     }
 
+    fun createLocalParticipantListCell(suffix: String) = ParticipantListCellModel(
+        localParticipantListCellStateFlow.value.displayName + suffix,
+        localParticipantListCellStateFlow.value.isMuted
+    )
+
     fun init(participantMap: Map<String, ParticipantInfoModel>, localUserState: LocalUserState) {
         val remoteParticipantList: List<ParticipantListCellModel> =
-            participantMap.values.map { ParticipantListCellModel(it.displayName, it.isMuted) }
+            participantMap.values.map {
+                ParticipantListCellModel(
+                    if (it.displayName.isEmpty())"Unnamed participant" else it.displayName,
+                    it.isMuted
+                )
+            }
         remoteParticipantListCellStateFlow = MutableStateFlow(remoteParticipantList)
         localParticipantListCellStateFlow =
             MutableStateFlow(
                 ParticipantListCellModel(
-                    localUserState.displayName,
+                    localUserState.displayName ?: "",
                     localUserState.audioState.operation == AudioOperationalStatus.OFF
                 )
             )
     }
 
     fun update(participantMap: Map<String, ParticipantInfoModel>, localUserState: LocalUserState) {
         val remoteParticipantList: List<ParticipantListCellModel> =
-            participantMap.values.map { ParticipantListCellModel(it.displayName, it.isMuted) }
+            participantMap.values.map {
+                ParticipantListCellModel(
+                    if (it.displayName.isEmpty())"Unnamed participant" else it.displayName,
```

Comment:
can be moved to view to get from resources, will help for localization?

Reply:
addressed